### PR TITLE
[NDS-338] Feat/Props Standardization (Booleans pt3)

### PR DESCRIPTION
**Pull request by Christos-Zacharopoulos**
## Description

**FOR V5**

This PR is part of a chain of PRs to follow regarding props globalization.

ticket: https://orfium.atlassian.net/browse/NDS-338

Also added the rule `"@typescript-eslint/naming-convention"` with the following configuration:
```
 "@typescript-eslint/naming-convention": [
      "error",
      { "selector": "variable", "types": ["boolean"],
        "format": ["PascalCase"],// As per documentation, prefix is trimmed before checking for format , so we need pascal case for values like `isBoolean`  https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#format-options
        "prefix": ["is", "has"] }
    ],
```

First, it complies with the general coding convention we have in the chapter and it also helps identify the places where the eslint rule about props is not working (forwarded components for example )

## Test Plan

Everything will be updated on a PR later. It's too much to keep track right now.

### CODE REVIEW

**Comment by Christos-Zacharopoulos** [comment]
@Orfium/front-end 
image

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -57,7 +58,10 @@ const Button = React.forwardRef<HTMLButtonElement, Props & ButtonProps>((props,
       </span>
     </ButtonBase>
   );
-});
+};
+
 Button.displayName = 'Button';
 
-export default Button;
+export default React.forwardRef<HTMLButtonElement, Props>((props, ref) => (
```
Comment (panvourtsis) [inline src/components/Button/Button.tsx:65]: Why we did the `forwardRef` here and not in the FC part :13

File: src/components/Radio/Radio.style.ts [hunk 2]
```diff
@@ -31,105 +31,108 @@ export const inputStyles: SerializedStyles = css`
...
-): SerializedStyles => css`
-  position: absolute;
-  border-radius: 50%;
-  width: ${rem('24px')};
-  height: ${rem('24px')};
-  transition: all 0.2s ease;
-  ${focused && !disabled && boxShadow({ colorScheme: theme.colorScheme })};
-`;
+export const customRadioInnerHover =
+  (focused: boolean, disabled: boolean) =>
```
Comment (panvourtsis) [inline src/components/Radio/Radio.style.ts:35]: these as well should be changed

File: src/components/Radio/Radio.style.ts [hunk 3]
```diff
@@ -31,105 +31,108 @@ export const inputStyles: SerializedStyles = css`
...
-): SerializedStyles => css`
-  position: relative;
-  border-radius: 50%;
-  width: ${rem('24px')};
-  height: ${rem('24px')};
-  transition: box-shadow 0.3s ease;
-  ${focused && !disabled && boxShadow({ colorScheme: theme.colorScheme })};
-`;
+export const customRadioWrapperStyles =
+  (focused: boolean, disabled: boolean) =>
```
Comment (panvourtsis) [inline src/components/Radio/Radio.style.ts:47]: here as well

File: src/components/Radio/Radio.style.ts [hunk 4]
```diff
@@ -31,105 +31,108 @@ export const inputStyles: SerializedStyles = css`
...
+        transform: translate(5px, 5px) scale(${props.isChecked ? '1' : '0'});
+        transition: transform 0.2s;
+      }
+    `;
+  };
 
-export const wrapperStyles = (disabled: boolean) => (theme: Theme): SerializedStyles => css`
-  position: relative;
+export const wrapperStyles =
+  (disabled: boolean) =>
```
Comment (panvourtsis) [inline src/components/Radio/Radio.style.ts:113]: `isDisabled`

File: src/components/Radio/Radio.tsx [hunk 5]
```diff
@@ -107,26 +111,28 @@ const Radio = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
...
       </span>
     </span>
   );
-});
+};
 
 Radio.displayName = 'Radio';
 
-export default Radio;
+export default React.forwardRef<HTMLInputElement, Props>((props, ref) => (
```
Comment (panvourtsis) [inline src/components/Radio/Radio.tsx:136]: Also about forwardRef here

**Review by panvourtsis — CHANGES_REQUESTED** [review]

**Comment by Christos-Zacharopoulos** [comment]
@panvourtsis 
All these issues were made due to a misconfiguration in the eslint rules regarding boolean prop naming. 

They will be resolved in another PR in order to not create merge conflicts to the chain of PRs.

**Pull request merged**